src/core/mod.rs

```diff
@@ -67,11 +68,20 @@ pub struct OscTrack {
 
 impl AvatarOsc {
     pub fn new(args: Args, multi: MultiProgress) -> AvatarOsc {
-        let ip = IpAddr::V4(Ipv4Addr::LOCALHOST);
+        // Parse the output IP address
+        let output_ip = match args.output_ip.parse::<IpAddr>() {
+            Ok(ip) => ip,
+            Err(_) => {
+                log::warn!("Invalid output IP address '{}', falling back to localhost", args.output_ip);
+                IpAddr::V4(Ipv4Addr::LOCALHOST)
```

Comment:
if user-provided arguments are wrong, just print an error and exit the program instead of disobeying the user.